### PR TITLE
Sets an environment variable when restarting due to preemption

### PR DESCRIPTION
**Pull request by StanfordMCP**

### CODE REVIEW

File: tony-core/src/main/java/com/linkedin/tony/ApplicationMaster.java [hunk 1]
```diff
@@ -353,6 +353,10 @@ private boolean run(String[] args) throws IOException {
       // Prepare for retryCount.
       reset();
       LOG.info("Retrying, remaining retry count" + amRetryCount);
+
+      //Set an environment variable when restarting due to preemption
+      containerEnv.put(Constants.AM_RETRY_COUNT, Integer.toString(amRetryCount));
```
Comment (oliverhu) [inline tony-core/src/main/java/com/linkedin/tony/ApplicationMaster.java:358]: This is the remaining retry count, we need the retry id (total-remaining) here.

File: tony-core/src/main/java/com/linkedin/tony/ApplicationMaster.java [hunk 2]
```diff
@@ -353,7 +354,13 @@ private boolean run(String[] args) throws IOException {
       // Prepare for retryCount.
       reset();
       LOG.info("Retrying, remaining retry count" + amRetryCount);
+
       amRetryCount -= 1;
+      numAMRetries += 1;
```
Comment (oliverhu) [inline tony-core/src/main/java/com/linkedin/tony/ApplicationMaster.java:359]: prefer to start from 0, move the +1 after next line

Comment (StanfordMCP) [reply tony-core/src/main/java/com/linkedin/tony/ApplicationMaster.java:359]: Ok, so set it to 0 within the scope of the method? Not line 118?

Comment (oliverhu) [reply tony-core/src/main/java/com/linkedin/tony/ApplicationMaster.java:359]: 118 should be ok, just move this to the nextline

**Comment by oliverhu** [comment]
👍  @StanfordMCP
 please also update the internal ticket with the details

**Comment by StanfordMCP** [comment]
Updated the internal ticket and closed it.

**Pull request merged**